### ISSUE TITLE
Flow type triggers space-infix-ops error

### ISSUE THREAD

**Issue by gajus** [babel/babel-eslint#226]
```js
(foo: string = 'test') => {};
```

I am getting an error:

> space-infix-ops: Infix operators must be spaced.

**Comment by adrienbrault** [comment]
Having this issue as well!

**Comment by gajus** [comment]
@hzoo Does this issue even fall in the `babel-eslint` domain?

**Comment by hzoo** [comment]
Unless babel-eslint is doing something wrong (possible), most likely the rule just doesn't account for flow correctly so you would have to overwrite the rule with the eslint-plugin-babel repo/rules.

**Comment by danez** [comment]
This is also reported at eslint/eslint#5211 and they also accept patches now for types.

**Comment by danez** [comment]
Fixed in eslint, and will be released probably with next version.

**Comment by OlegZee** [comment]
My package references "babel-eslint": "^6.1.2" and I updated eslint to 3.2.2 but still getting the error.
Why this issue is closed?

**Comment by danez** [comment]
Because it should have been fixed in eslint 3.2, I'm gonna check

**Comment by OlegZee** [comment]
What a great response time?!

**Comment by OlegZee** [comment]
@danez 
eslint works fine, the problem is in outdated gulp-eslint task which uses eslint@2.1.0.
Thank you for quick reply and sorry for troubling you.

**Comment by emirotin** [comment]
I have the same issue with the most recent versions of everything. Minimal repro: https://github.com/emirotin/flow-lint-issue

**Comment by danez** [comment]
@emirotin Thanks for making the example repo, I confirmed that polymorphic functions cause this error. Created a new Issue #366
